tools/resourcedocsgen/pkg/util/language/language.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package language exposes an enumeration of docs generation target languages:
+//
+// - [CSharp]
+// - [Go]
+// - [Java]
+// - [Python]
+// - [Typescript]
+// - [YAML]
+//
+// For a list of all languages, use [ListAll].
+package language
+
+// Language is a valid docsgen language. The default value is not valid.
+type Language struct{ tag string }
+
+var (
+	CSharp     = Language{"csharp"}
+	Go         = Language{"go"}
+	Java       = Language{"java"}
+	Python     = Language{"python"}
+	Typescript = Language{"nodejs"}
+	YAML       = Language{"yaml"}
+)
+
+// Iteration order of a [Set] of languages is well defined.
+//
+// When-ever a new language is added above, it needs to be added here.
+var langOrder = [...]Language{
+	CSharp,
+	Go,
+	Typescript,
+	Python,
+	YAML,
+	Java,
+}
```

Comment:
Another thing it feels like we could kill (and in the process simplify things should anyone ever need to add a new language) if we get rid of `Set`.

Reply:
We won't really be able to kill this, since we need to have *a* deterministic iteration order for languages.